### ISSUE TITLE
Suggestions

### ISSUE THREAD

**Issue by TheSaltySeagull** [BelegCufea/CompassHUD#6]
I have a couple of suggestions that may or may not be enhancements:

1. Ticks need a vertical adjustment setting. [This looks wrong](https://bit.ly/4fRVfUd).
2. The Pin that marks the center needs that as well. (I'd suggest that the height of your frame can be adjusted. When someone wants to use a big font size it'll clip for example.)
3. Add LibSharedMedia support for your Border and Backgrounds. The available options are, at least to me, not viable options.
4. I'd love to be able to adjust the width to my full screen width (that can be scrapped if you would add LibSharedMedia support, because then a Background that fades out would counter the hard edge when turning)

again, this is the best addon I've discovered so far that is enhancing my immersive RPG feeling!

**Comment by BelegCufea** [comment]
As for the LibShared Media, I have it already, both for borders and backgrounds.
![image](https://github.com/user-attachments/assets/6850ff67-c491-4da9-9a4f-d1813ca8eecd)¨

Are you looking for textures maybe? (screen shot form anther addon).
![image](https://github.com/user-attachments/assets/002982b2-3a09-482b-9fb2-e4ece3a2f1f7)

Will see what I can do about the size adjustments. It is hardcoded now as it worked originaly only with predefined image.

I don't know what you mean by vertial adjustment of ticks and center pin mark? They are both at the edge (top/bottom/ of the HUD. You want to move them outside like the pointers?

**Comment by TheSaltySeagull** [comment]
[Here's what I've done with WA](https://bit.ly/479GFUt) [little video test](https://youtu.be/TBxRwsQVGxs). I'm not happy with the texture quite yet but this is where it eases out the compass more, so it doesn't cut so hard at the edges when turning.
So, I might have been a little too forward with the LibSharedMedia support, and yes, I meant textures 😜 (my bad)

So, maybe these are better:

1. That center Pin should be able to be disabled. It kinda looks off. I understand the need for it, but it doesn't really fit well overall.
2. I don't quite understand the ability to vertically align the Text when it is able to move off frame., or is it intentionally for this reason? https://bit.ly/3AxxhO7
3. Same with Font Size.  https://bit.ly/4czZaT0
4. Forget what I suggested about the ticks, I can actually get behind it. My issue is the bigger font size I use which makes it look odd, but honestly it isn't that bad.

**Comment by BelegCufea** [comment]
1. Will do. Maybe in future I will make the texture to be configurable.
2 and 3. Some fonts are positioned lower or higher or are too small or to large for internal logic to handle it, so I put the options there.

As for the width. I am working on custom scaling in X and Y.

As for the texture. There is option to make your own texture for the HUD. The process is mentioned in TODO section of ReadMe.

**Comment by BelegCufea** [comment]
![image](https://github.com/user-attachments/assets/77abd254-aee6-4c8e-9c32-fd71239292e8)

![image](https://github.com/user-attachments/assets/d4e7bbb3-a890-45a6-b8a7-c0d02ac122a9)

https://github.com/BelegCufea/CompassHUD/tree/ScaleHUD

If you will be so kind and test this. It may even help with #5

**Comment by BelegCufea** [comment]
I was able to do this with already build-in options:

![image](https://github.com/user-attachments/assets/d8f9d0c4-bd38-46c9-821d-e1e817b5c28c)

I just made new backkgroud:
[CHbg1.zip](https://github.com/user-attachments/files/16673461/CHbg1.zip)

I have put it into Media directory of my addon and registered it with LibSharedMedia:

```lua
local LSM = LibStub("LibSharedMedia-3.0")
LSM:Register(LSM.MediaType.BACKGROUND, "MyCustomBackground", [[Interface\Addons\CompassHUD\Media\CHbg1.tga]])

```
Picked it in backgrounds and also edit the Edge lines a bit:
![image](https://github.com/user-attachments/assets/87288f71-5de9-42b7-a680-76ec67e0bbc7)

Edit: the white one is even better coz you can change colors:
[CHbg2.zip](https://github.com/user-attachments/files/16673661/CHbg2.zip)

**Comment by raptormama** [comment]
I love the idea of making the center tick more visible. I have my options set to resemble the compass tape of many aircraft heads-up displays, similar to [this one](https://ssl.c.photoshelter.com/img-get2/I0000vKyKfgWmc8s/fit=1000x750/RB-0160.jpg) (a C-130J Hercules flying a heading of 270, due west) -- including the green phosphor colour you see here.

It would be a nice bonus to be able to display the current heading above the pointer, as is the case in that image, but that might be a bit of work. Even a simple downward-pointing triangle above the heading "tape", with changeable colour as you propose, would be more visible for me.

**Comment by BelegCufea** [comment]
You mean to display the degree number up there? I can add that as an option. It would be nice, especially if you have a plain texture without the degrees or even the cardinal directions.

I’ve also been thinking about the center pin. Originally, I wanted to let the player choose a texture, but I'm not sure if you can change its color. I’ll try a few things and see what works...

**Comment by BelegCufea** [comment]
As for the heading feature, I've got something set up.

The options are currently just thrown together—I’ll clean them up once all the functionality is in place.

The code is available in the [Heading branch](https://github.com/BelegCufea/CompassHUD/tree/Heading).

![image](https://github.com/user-attachments/assets/aab855b9-03b5-41c1-8c48-30c05525a3f4)

If you have time, I would really appreciate it if you could test it. As always, any suggestions are welcome!

**Comment by raptormama** [comment]
So far so good! I have encountered no Lua errors, and the settings work fine. I'm including a screenshot of my full UI and a screenshot of how I set up my heading. You might enjoy getting a look at them.

![compasshud-heading-in-flight](https://github.com/user-attachments/assets/46bf4fc9-f7a7-4af7-aac1-381dd0b84b05)

![compasshud-heading-settings](https://github.com/user-attachments/assets/983bb7c4-6afc-43aa-a692-e0a2ef5e4aa2)

One caveat that is not your fault: it's possible for the heading display to display both 0 and 360 for due north, so it has two 'ticks' that mean the same thing.. However, aircraft compasses will display 360 instead of 0 because simply stating "zero" (pronounced "zee-ro") can be ambiguous, and "three six zee-ro" is easier to comprehend in voice communications than just a singular "zee-ro" and is faster to say than repeating "zee-ro" three times if the heading were displayed as 000.

So, the heading code should automatically display 360 whenever it would ordinarily show 0. I've been able to get it to do so several times, though probably most people wouldn't test this. They are an equivalent reading.

Some people may prefer this; some may not. Perhaps there could be a checkbox for "display north as 360" and default to displaying as 0.

My guess is that the reason that both 360 and 0 can appear is due to the way Lua rounds the result it receives.
